metaflow/cli.py

```diff
@@ -606,11 +694,124 @@ def run(obj,
     write_latest_run_id(obj, runtime.run_id)
     write_run_id(run_id_file, runtime.run_id)
 
+    # import pickle
+    #
+    # with open('graph.pkl', 'wb') as gh:
+    #     pickle.dump(obj.graph, gh)
+
     parameters.set_parameters(obj.flow, kwargs)
     runtime.persist_parameters()
     runtime.execute()
 
 
+@parameters.add_custom_parameters
+@cli.command(help='Run the initial part of the workflow (i.e., save parameters locally).')
+@common_run_options
+@click.option('--namespace',
+              'user_namespace',
+              default=None,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag. Note that this option does not alter "
+                   "tags assigned to the objects produced by this run, just "
+                   "what existing objects are visible in the client API. You "
+                   "can enable the global namespace with an empty string."
+                   "--namespace=")
+@click.pass_obj
+def pre_start(obj,
+        tags=None,
+        max_workers=None,
+        max_num_splits=None,
+        max_log_size=None,
+        decospecs=None,
+        run_id_file=None,
+        user_namespace=None,
+        **kwargs):
+
+    if namespace is not None:
+        namespace(user_namespace or None)
+    before_run(obj, tags, decospecs + obj.environment.decospecs())
+
+    runtime = NativeRuntime(obj.flow,
+                            obj.graph,
+                            obj.datastore,
+                            obj.metadata,
+                            obj.environment,
+                            obj.package,
+                            obj.logger,
+                            obj.entrypoint,
+                            obj.event_logger,
+                            obj.monitor,
+                            max_workers=max_workers,
+                            max_num_splits=max_num_splits,
+                            max_log_size=max_log_size * 1024 * 1024)
+    write_latest_run_id(obj, runtime.run_id)
+    write_run_id(run_id_file, runtime.run_id)
+
+    parameters.set_parameters(obj.flow, kwargs)
+    runtime.persist_parameters()
+
+    # FORMAT: ($1)datastore_root location \t ($2)run_id \t ($3)next_step_to_run \t ($4)task_id(of next step) \t ($5)current step \t ($6)task_id(of current step)
+    print(f"{obj.datastore.datastore_root}\t{runtime.run_id}\tstart\t1\t_parameters\t0")
```

Comment:
what are we printing here?
Also, this uses python `f` strings.  Does the rest of the codebase use f strings, does Metaflow need to support python 2?  Do all version of Python 3 that Metaflow uses support f strings?

Reply:
I'm using this print to be able to perform the orchestration outside of Metaflow (the part where I use AWK to construct the command for the next step to be executed). This could be removed once we are able to save state. (I have now added a TODO here to note that as well)